### PR TITLE
Adding shadow of AccessibilityNodeInfo and AccessibilityNodeInfoCompat

### PR DESCRIPTION
**Pull request by zhongyu05**
Continuing to add shadows of accessibility APIs.

Also bumping version of android support library to 22 in order to build newly introduced apis in 
AccessibilityNodeInfoCompat.

### CODE REVIEW

**Comment by zhongyu05** [comment]
Travis is not building because it doesn't have the latest android support library 22.2.0 under $ANDROID_HOME/extra/support/v4, it can be solved by updating its sdk in before_install scripts. (e.g. android update sdk --no-ui --filter extra-android-support)

Not I am not using the library under $ANDORID_HOME/extras/android/m2repository/com/android/support/support-v4/22.2.0/ for the following reason:

After 19.0.1 Google starts to publish support library under m2repository with aar format. But installing those aars do not have all the classes in the latest jar under extra-android-support. If I install one of these into local maven repo and build robolectric against it, runtime errors (Noclassdeffounderror, etc.) will be introduced when running tests.

**Comment by erd** [comment]
> After 19.0.1 Google starts to publish support library under m2repository with aar format. But installing those aars do not have all the classes in the latest jar under extra-android-support.

I ran into the same issue when trying to fix #1830. Let me see if I can get Robolectric compiling correctly with support-v4 22.2.0 before I merge this.

**Comment by zhongyu05** [comment]
Thanks Erich, let me know if you can make support-v4-22.2.0 work then I'll modify my pull request.

Otherwise, I think it's better to fix travis before merging.

**Comment by zhongyu05** [comment]
@erd, I realized I can just change the .travis.yml file and make it build, can we merge it now?

**Comment by erd** [comment]
I'd like to try it out before merging and I'm also going to squash all the commits into one (since there are several wip commits).

File: scripts/install-dependencies.sh [hunk 1]
```diff
@@ -71,7 +71,7 @@ if [ -z "${ANDROID_HOME}" ]; then
 fi
 
 # Install support-v4
-install_jar "com.android.support" "support-v4" "19.0.1" "${ANDROID_REPOSITORY}/com/android/support/support-v4/19.0.1/support-v4-19.0.1.jar"
+install_jar "com.android.support" "support-v4" "22.2.0" "$ANDROID_HOME/extras/android/support/v4/android-support-v4.jar"
```
Comment (erd) [inline scripts/install-dependencies.sh]: I'm a little hesitant to install the jar from the un-versioned source. Meaning, the jar could get updated to 23.0.0 and we'd still be installing it as version 22.2.0 (which could impact projects that use robolectric). Did you ever figure out why there are classes missing from the aar that are present in the jar?

Comment (zhongyu05) [inline scripts/install-dependencies.sh]: Good point, I don't know the answer right now, will see if I can find the right person to ask.

**Comment by zhongyu05** [comment]
Hi @erd, figured out why the aar is incomplete, there is actually an internal_impl-22.2.0.jar inside the libs folder of support-v4-22.2.0.aar. We need to install that as well, now it should all be good.

**Comment by erd** [comment]
@zhongyu05 Awesome!

**Comment by zhongyu05** [comment]
Thanks for the correction, some of those setters/getters might be redundant. I'll do another pass to remove unnecessary methods, equals have different logic than the underlying class so it is needed.

**Comment by zhongyu05** [comment]
Hi @erd , I have cleaned up the files and added some additional action support. I think it's ready.

**Pull request merged**